src/legacy/core_plugins/data/public/shim/legacy_module.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { once } from 'lodash';
+
+import { wrapInI18nContext } from 'ui/i18n';
+import { Filter } from '@kbn/es-query';
+
+// @ts-ignore
+import { uiModules } from 'ui/modules';
+import { FilterBar } from '../filter';
+import { ApplyFiltersPopover } from '../filter';
+import template from './apply_filter_directive.html';
+
+// @ts-ignore
+import { mapAndFlattenFilters } from '../filter/filter_manager/lib/map_and_flatten_filters';
+// @ts-ignore
+
+/** @internal */
+export const initLegacyModule = once((): void => {
+  uiModules
+    .get('app/kibana')
+    .directive('filterBar', (reactDirective: any) => {
+      return reactDirective(wrapInI18nContext(FilterBar));
+    })
+    .directive('applyFiltersPopoverComponent', (reactDirective: any) => {
+      return reactDirective(wrapInI18nContext(ApplyFiltersPopover));
+    })
+    .directive('applyFiltersPopover', (indexPatterns: any) => {
```

Comment:
nit: should be able to use `IndexPattern` or `StaticIndexPattern` here now that the other PR has merged :)

Reply:
But which one should I use?
Still not sure about this!

Reply:
sorry, I meant `IndexPatterns` -- that should be what you are looking for here. sorry for the confusion 😄